src/app/[locale]/guides/data-validation/en.mdx

```diff
@@ -0,0 +1,54 @@
+export const metadata = {
+  title: 'Data Validation',
+  description:
+    'Expectations around recommended data limits and validation',
+}
+
+# Data Validation
+
+Some software does not need to worry much about Bluesky or atproto schema validation. For example, client apps and bots can generally assume that data they receive from the Bluesky API is valid, and when they try to create records the PDS is responsible for double-checking schema validation.
+
+Software which consumes directly from the event stream (firehose) should be more careful. And developers designing new tools or clever hacks should be aware of some expectations and hard limits around data validation.
+
+If an individual record fails to validate for any reason, the entire record should be ignored, but other records from the same repository should be processed. If there is a problem with the repository commit data or repository structure ("MST"), the repository should be marked as invalid, but old content does not need to be de-indexed or purged. As soon as a new valid commit is received, the repository should be marked valid again. This logic mostly relevant to feed generators, AppViews, and moderation services, all of which may be consuming from the firehose.
+
+
+## Recommended Data Limits
+
+The following are informal guidelines, mostly to communicate order-of-magnitude expectations, and are likely to evolve over time. They are not part of the atproto specification, but if you are pushing the limits here, you may limit interoperability.
+
+**CBOR Record Size:** try to keep individual records to a few dozen KBytes. If you need to store more data, even text data, consider using a blob instead. A reasonable maximum record size limit (`MAX_CBOR_RECORD_SIZE`) is 1 MByte.
+
+Note that event stream (firehose) "frames" may consist of multiple records, and larger limits are recommended for CBOR parsing in general, on the order of 4-5 MBytes.
```

Comment:
I assume we're referencing the `sync.subscribeRepos` endpoint here.  There is a cap of 1mb in the `blocks` field, so indeed creation of a 1mb record would end-up being marked as `tooBig`.

Reply:
hmm yeah it may be worth bumping that?

Reply:
please bump it 1mb is atrocious :p

Reply:
The current subscribeRepos blocks byte limit is 2,000,000 (power of 10, not mebibytes). I added a comment about this here.